src/java/ognl/OgnlRuntime.java

```diff
@@ -2076,11 +2076,26 @@ public static Object getStaticField(OgnlContext context, String className, Strin
                 }
             }
 
-            Field f = c.getField(fieldName);
+            Field f = getField(c, fieldName);
+            if (f == null) {
+                throw new NoSuchFieldException(fieldName);
+            }
             if (!Modifier.isStatic(f.getModifiers()))
                 throw new OgnlException("Field " + fieldName + " of class " + className + " is not static");
 
-            return f.get(null);
+            Object result = null;
+            if (context.getMemberAccess().isAccessible(context, null, f, null)) {
+                Object state = context.getMemberAccess().setup(context, null, f, null);
+                try {
+                    result = f.get(null);
+                } finally {
+                    context.getMemberAccess().restore(context, null, f, null, state);
+                }
+            } else {
+                result = f.get(null);
```

Comment:
Now I wonder if we shouldn't throw an exception here, I mean, the field is not accessible but we want to get access to it anyway.

Reply:
@lukaszlenart  Do you mean line 2079? 

```java
            Field f = getField(c, fieldName);
            if (f == null) {
                throw new NoSuchFieldException(fieldName);
            }
```

`OgnlRuntime#getField` return all fields, contain private fields, so throw `NoSuchFieldException` is ok.

Reply:
No, I meant that we check access first and if the field is not accessible we access it anyway. This can lead to a security vulnerability as basically we ignore if the field is accessible or not.

Reply:
@lukaszlenart I got it, I updated the commit.